### DISCUSSION TITLE
very slow to do anything after update?

### DISCUSSION THREAD

**Discussion by lgwjames** [lllyasviel/Fooocus#2641] [Q&A]
normally i run thermal and within 2 mins all loaded and ready to go

been sat like this for 10 mins since updating

Last login: Tue Mar 26 18:43:49 on ttys000
You have mail.
/Users/Quad-Core/Desktop/fooocus.command ; exit;
(base) Quad-Core@iMac ~ % /Users/Quad-Core/Desktop/fooocus.command ; exit;
Already up-to-date
Update succeeded.
[System ARGV] ['entry_with_update.py', '--theme', 'dark']
Python 3.11.5 (main, Sep 11 2023, 08:19:27) [Clang 14.0.6 ]
Fooocus version: 2.3.1
[Cleanup] Attempting to delete content of temp dir /var/folders/6l/xrkxgs657g7_vq50479x7lc40000gp/T/fooocus
[Cleanup] Cleanup successful
Total VRAM 65536 MB, total RAM 65536 MB
Set vram state to: SHARED
Always offload VRAM
Device: mps
VAE dtype: torch.float32
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --attention-split
Refiner unloaded.
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
model_type EPS
UNet ADM Dimension 2816
Using split attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using split attention in VAE

then after another 5 mins

extra {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.text_projection'}
Base model loaded: /Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors
Request to load LoRAs [['JessicaL.safetensors', 1.0], ['PerfectEyesXL.safetensors', 1.0], ['Perfect Hands v2.safetensors', 1.0], ['None', 1.0], ['None', 1.0]] for model [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors].
Loaded LoRA [/Volumes/Ai works/Loras/PerfectEyesXL.safetensors] for UNet [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
Loaded LoRA [/Volumes/Ai works/Loras/PerfectEyesXL.safetensors] for CLIP [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
Loaded LoRA [/Volumes/Ai works/Loras/Perfect Hands v2.safetensors] for UNet [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
Loaded LoRA [/Volumes/Ai works/Loras/Perfect Hands v2.safetensors] for CLIP [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cpu, use_fp16 = False.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 7.72 seconds
Started worker with PID 9294
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 5532741127733980317
[Fooocus] Downloading control models ...
[Fooocus] Loading control models ...
extra clip vision: ['vision_model.embeddings.position_ids']
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
Request to load LoRAs [['PerfectEyesXL.safetensors', 1.0], ['Perfect Hands v2.safetensors', 1.0], ['None', 1.0], ['None', 1.0]] for model [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors].
Loaded LoRA [/Volumes/Ai works/Loras/PerfectEyesXL.safetensors] for UNet [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
Loaded LoRA [/Volumes/Ai works/Loras/PerfectEyesXL.safetensors] for CLIP [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
Loaded LoRA [/Volumes/Ai works/Loras/Perfect Hands v2.safetensors] for UNet [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
Loaded LoRA [/Volumes/Ai works/Loras/Perfect Hands v2.safetensors] for CLIP [/Volumes/Ai works/Checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
Requested to load SDXLClipModel
Loading 1 new model
unload clone 1
[Fooocus Model Management] Moving model(s) has taken 3.39 seconds
[Fooocus] Processing prompts ...

any ideas?

**Comment by lgwjames** [comment]
today 

Last login: Thu Mar 28 17:06:36 on ttys000
You have mail.
/Users/Quad-Core/Desktop/fooocus.command ; exit;
(base) Quad-Core@iMac ~ % /Users/Quad-Core/Desktop/fooocus.command ; exit;
Already up-to-date
Update succeeded.
[System ARGV] ['entry_with_update.py', '--theme', 'dark']
Python 3.11.5 (main, Sep 11 2023, 08:19:27) [Clang 14.0.6 ]
Fooocus version: 2.3.1
[Cleanup] Attempting to delete content of temp dir /var/folders/6l/xrkxgs657g7_vq50479x7lc40000gp/T/fooocus
[Cleanup] Cleanup successful

after 3 mins

Total VRAM 65536 MB, total RAM 65536 MB
Set vram state to: SHARED
Always offload VRAM
Device: mps
VAE dtype: torch.float32
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --attention-split

loads GUI but still not loading fully yet

**Reply by Rogala** [reply]
There may be a problem with your version of python, by default I have 3.10.9 and you already have 3.11.5, python has a bad property of not working well when changing version.
But this is not exact check.

[You have a macOS, read it, maybe you missed something?](https://github.com/lllyasviel/Fooocus#mac)

**Comment by lgwjames** [comment]
was working ok before the last few updates, how do i downgrade

**Comment by lgwjames** [comment]
followed here, https://saturncloud.io/blog/downgrading-python-version-using-conda-a-guide/
says i have 3.9 but when run fooocus still says 3.11.5

**Reply by Rogala** [reply]
No help, I don't work with macOS.
Try changing the Fooocus version to 2.3.0 if that version worked fine.
From the command line in the Fooocus folder, type in
`git checkout 2.3.0`

**Comment by lgwjames** [comment]
and then remove the update part from run ?

**Reply by Rogala** [reply]
No, it will roll back to the previous version.

**Comment by lgwjames** [comment]
gone back to 2.2.1 and even moved some files to a diff drive, as i point to a external SSD as my mac one is low, 45gb free and a lot of loras loaded and faster to load then it was for last few days, 41.7s/it still taking 21 mins a image, used to be 12-15

**Comment by lgwjames** [comment]
just a update, still problems, 3-7 mins to load my config file of my loras , setting etc 
tried 2.21 up to 2.30 
even moved drives and the same

now i get

Update failed.
'refs/heads/HEAD'
Update succeeded.
[System ARGV] ['entry_with_update.py', '--disable-offload-from-vram', '--theme', 'dark']
Python 3.11.5 (main, Sep 11 2023, 08:19:27) [Clang 14.0.6 ]
Fooocus version: 2.3.1
